src/graph/views.tsx

```diff
@@ -140,10 +235,10 @@ export class SLabelView extends ShapeView {
 
 @injectable()
 export class SCompartmentView implements IView {
-    render(compartment: Readonly<SCompartment>, context: RenderingContext): VNode | undefined {
+    render(compartment: Readonly<SCompartment>, context: RenderingContext, args?: object): VNode | undefined {
         const translate = `translate(${compartment.bounds.x}, ${compartment.bounds.y})`;
         const vnode = <g transform={translate} class-sprotty-comp="{true}">
-            {context.renderChildren(compartment)}
+            {context.renderChildren(compartment, args)}
```

Comment:
The necessity to explicitly pass the `args` through all views is a change that needs to be explained, at least in the CHANGELOG.

I'm thinking whether we should make more explicit how the parameter is used. Maybe make the `args` parameter mandatory? Or define a more concrete type than `object`?

Reply:
Yeah, good point! We should definitely mention that. I'll add this to the readme.

Whether to really enforce passing arguments or not: I think I'd be in favor to make the recommendation of passing those arguments on to the rendering of its children more explicit. However, as it'll break practically all existing `IView` implementations I'm a bit hesitant whether it is worth it and whether a recommendation in the Readme wouldn't be enough. What do you think?

Reply:
What do you think about this idea:
 * Add the following property to `RenderingContext` and to the constructor of `ModelRenderer`:
```typescript
readonly parentArgs?: object;
```
 * Change `renderChildren` to this:
```typescript
const context = args === undefined ? this : new ModelRenderer(this.viewRegistry, this.targetKind, this.postprocessors, args);
return element.children
    .map(child => context.renderElement(child, args))
    .filter(vnode => vnode !== undefined) as VNode[];
```
 * Change `renderElement` to this:
```typescript
const view = this.viewRegistry.get(element.type);
const vnode = view.render(element, this, this.parentArgs);
```
We could also consider merging the new `args` with the previous `parentArgs` to support properties reaching through to nested elements even when another view adds more properties.

And to make the whole thing more type-safe and be more consistent with React, we could renams `args` to `props` and change `IView` to this:
```typescript
export interface IView<P = {}> {
    render(model: Readonly<SModelElement>, context: RenderingContext, P: object): VNode | undefined
}
```
AFAICT these changes would be backwards-compatible.

Reply:
On a first glance this feels like a very good idea! Thanks!

Another idea instead of directly merging the `args` could be to always add them as `parentArgs` (args = `{...args, parentArgs: this.parentArgs}`)? This would most likely avoid unexpected property name clashes and consumers of the args could recursively query specific properties (such as the pre-computed routes which are on the root `args`)? Sprotty could provide a util function for that, similar to `findParentByFeature`.

Regarding the `props`... I like the idea of the generic type making the args more typesafe. Regarding the rename to `props`, I don't have a hard opinion, but I feel like the `model` corresponds to react's concept of `props` a bit more than `args`.

From your suggestions I took that you actually already have those changes locally, right? Do you want to add a commit to this PR or should I work on that?

Reply:
> Another idea instead of directly merging the args could be to always add them as parentArgs (args = {...args, parentArgs: this.parentArgs})? This would most likely avoid unexpected property name clashes and consumers of the args could recursively query specific properties (such as the pre-computed routes which are on the root args)? Sprotty could provide a util function for that, similar to findParentByFeature.

Ok that seems like the most robust solution, but it needs a utility function to get the root arguments. Let's go for it.

> Regarding the props... I like the idea of the generic type making the args more typesafe. Regarding the rename to props, I don't have a hard opinion, but I feel like the model corresponds to react's concept of props a bit more than args.

Ok let's keep `args` for now.

> From your suggestions I took that you actually already have those changes locally, right? Do you want to add a commit to this PR or should I work on that?

No I don't have any local changes, the proposals were more like a scratchpad.